File: 05 SWEA/D1/Java/src/Prob1933.java

/**
* title  : 1933. 간단한 N의 약수 (D1)
* time   : 104ms
* memory : 18900kb
*/

import java.io.*;
 
public class Prob1933 {
    
    public static void main(String[] args) throws IOException {

        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        StringBuilder sb = new StringBuilder();

        int N = Integer.parseInt(br.readLine());

        for (int i = 1; i <= N; i++) {
            if (N % i == 0) {
                sb.append(i).append(" ");
            }
        }

        System.out.println(sb);
    }
}

File: 05 SWEA/D1/Python/Prob1933.py

"""
title  : 1933. 간단한 N의 약수 (D1)
time   : 106ms
memory : 44780kb
"""

def solution():

    N = int(input())
    for idx in range(1, N//2+1):
        if (N % idx == 0):
            print(idx, end=" ")
    
    print(N)

solution()